psi4/src/psi4/libmints/matrix.h

```diff
@@ -852,14 +852,6 @@ class PSI_API Matrix : public std::enable_shared_from_this<Matrix> {
     /// Diagonalizes this, eigvectors and eigvalues must be created by caller.  Only for symmetric matrices.
     void diagonalize(Matrix* eigvectors, Vector* eigvalues, diagonalize_order nMatz = ascending);
     void diagonalize(SharedMatrix& eigvectors, std::shared_ptr<Vector>& eigvalues, diagonalize_order nMatz = ascending);
```

Comment:
The `diagonalize` function should take references and not bother with pointers at all.